crates/node/src/storage/mod.rs

```diff
@@ -544,6 +563,41 @@ mod tests {
             }
         }
 
+        // Check NoDiff subtree
+        let mut current_state = HashMap::new();
+        for i in 0..num_keys {
+            let key = make_key(i);
+            current_state.insert(key, true);
+        }
+        for (height, key, write_type) in blocks_write_type {
+            if key != client_counter_key() {
+                continue;
+            }
+            let merkle_key =
+                Key::from(NO_DIFF_KEY_PREFIX.to_string().to_db_key())
+                    .join(&key);
+            let tree =
+                state.get_merkle_tree(height, Some(StoreType::NoDiff))?;
+            assert_eq!(tree.root().0, roots.get(&height).unwrap().0);
+            match write_type {
```

Comment:
Needs comments, this logic is not clear.

Reply:
added a comment cd441a2

Reply:
Can you explain the semantics of these 0/1/3 write types?

Reply:
44c9201